[DMS-77] Add usage examples to the documentation

s-and-witch opened the pull request.

GoPavel commented:
I think it's better to add time and space complexity for `fromIter`, `put`, `delete` etc.
For some of them, it's obvious and for others, it's a repetition of what was written at the beginning of the doc page (in the overview), but I think it will be easier for a reader to have it repeated next to each function.

GoPavel commented on src/PersistentOrderedMap.mo:
```diff
@@ -102,7 +251,18 @@ module {
+  /// Debug.print(debug_show(Iter.toArray(Map.iter(rbMap, #fwd))));
+  /// Debug.print(debug_show(Iter.toArray(Map.iter(rbMap, #bwd))));
+  ///
+  /// //  stdout [(0, "Zero"), (1, "One"), (2, "Two")]
+  /// //  stdout [(2, "Two"), (1, "One"), (0, "Zero")]
```
```suggestion
  /// //  [(0, "Zero"), (1, "One"), (2, "Two")]
  /// //  [(2, "Two"), (1, "One"), (0, "Zero")]
```

s-and-witch replied:
Fixed

GoPavel commented:
Also a couple of high-level suggestions about documentation:
+ I think it would be good to mention the stability property somewhere in the overview (at the beginning)
+ It would be good to add some words about `MapOps` in its own doc section or in the overview.

s-and-witch commented:
@GoPavel updated patch according to your suggestions
- Added complexity information everywhere
- Mentioned, that it's stable (perhaps we may be even more explicit, but I don't know what to write there)
- Added a couple of words about `MapOps` and that it contains operations that either reshape or query `Map`.
